### ISSUE TITLE
`{showtext}` does not work with `{pkgdown}`

### ISSUE THREAD

**Issue by thomas-neitmann** [yixuan/showtext#36]
I recently decided to use `{showtext}` in my [`{ggcharts}`](https://github.com/thomas-neitmann/ggcharts) package. I created an [`onLoad()`](https://github.com/thomas-neitmann/ggcharts/blob/master/R/zzz.R) function which adds my fonts and calls `showtext_auto()`. When loading the package the fonts appear as expected. However, when creating my package website with `{pkgdown}` the fonts do not get rendered. Instead I get warnings about the font not being available (similar to when using `{showtext}` in RStudio). Running this on GitHub actions even throws an error (see [here](https://github.com/thomas-neitmann/ggcharts/runs/702210034?check_suite_focus=true)). Any ideas why this could be the case?

**Comment by yixuan** [comment]
Not sure how to handle the references, but for vignettes/articles, just pass the `fig.showtext = TRUE` option to `knitr`:

```r
knitr::opts_chunk$set(
  message = FALSE,
  warning = FALSE,
  fig.width = 7,
  fig.height = 5,
  fig.showtext = TRUE
)
# You may also need to adjust the DPI here
showtext::showtext_opts(dpi = 120)
```

**Comment by thomas-neitmann** [comment]
Currently, `{pkgdown}` does not support setting any options for the examples. I found a workaround, though. I added the following code to my `.onLoad()` function:
```r
  if (Sys.getenv("IN_PKGDOWN") == "true") {
    showtext::showtext_opts(dpi = 192)
    trace(
      what = grDevices::png,
      exit = showtext::showtext_begin,
      print = FALSE
    )
  }
```
This is basically using the same workaround to make `{showtext}` work with the RStudio graphics device. The code only runs if the package is loaded when `{pkgdown}` is running. I had to set the DPI to 192 since `{pkgdown}` sets the `fig.retina` options to 2 which means that the default DPI of 96 gets multiplied by 2.

Unfortunately, I could only manage to make this work when running `pkgdown::deploy_to_brach()` locally. When running it on GitHub actions it seems the package is already loaded before `pkgdown::deploy_to_branch()` is called and thus the `if` statement isn't run.
